packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.base.tsx

```diff
@@ -194,11 +196,29 @@ export class GroupedListBase extends BaseComponent<IGroupedListProps, IGroupedLi
     return 1;
   }
 
-  private _getPageHeight = (getGroupHeight: (group?: IGroup, groupIndex?: number) => number) => (itemIndex: number) => {
-    const { groups } = this.state;
+  private _getDefaultGroupItemLimit = (group: IGroup): number => {
+    return group.count;
+  };
+
+  private _getGroupItemLimit = (group: IGroup): number => {
+    const { groupProps } = this.props;
+    const getGroupItemLimit = groupProps && groupProps.getGroupItemLimit ? groupProps.getGroupItemLimit : this._getDefaultGroupItemLimit;
+
+    return getGroupItemLimit(group);
+  };
 
+  private _getGroupHeight = (group: IGroup): number => {
+    const rowHeight = this.props.compact ? COMPACT_ROW_HEIGHT : ROW_HEIGHT;
+
+    return rowHeight + (group.isCollapsed ? 0 : rowHeight * this._getGroupItemLimit(group));
+  };
+
+  private _getPageHeight: IListProps['getPageHeight'] = (itemIndex: number) => {
+    const { groups } = this.state;
+    const { getGroupHeight = this._getGroupHeight } = this.props;
     const pageGroup = groups && groups[itemIndex];
-    return getGroupHeight(pageGroup, itemIndex);
+
+    return getGroupHeight(pageGroup!, itemIndex);
```

Comment:
Is the non-null assertion necessary?

Reply:
Great call! When writing this I really wanted to avoid this non-null assertion. I think we can avoid it with the following refactor:

```diff
diff --git a/packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.base.tsx b/packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.base.tsx
index 947364053..bbd6834cd 100644
--- a/packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.base.tsx
+++ b/packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.base.tsx
@@ -218,7 +218,11 @@ export class GroupedListBase extends BaseComponent<IGroupedListProps, IGroupedLi
     const { getGroupHeight = this._getGroupHeight } = this.props;
     const pageGroup = groups && groups[itemIndex];
 
-    return getGroupHeight(pageGroup!, itemIndex);
+    if (pageGroup) {
+      return getGroupHeight(pageGroup, itemIndex);
+    } else {
+      return 0;
+    }
   };
 
   private _getGroupKey(group: IGroup | undefined, index: number): string {
```

It seems safe given that the absence of a group _should_ mean that the inner list is of height zero. cc: @cliffkoh for his thoughts re: avoiding the assertion.

Reply:
Tested and the above works well, committed in https://github.com/OfficeDev/office-ui-fabric-react/pull/8125/commits/f8326840ac0255e5ec367cc7b9ec8afd3ecfb5b3